### DISCUSSION TITLE
How to connect to the postgresql database?

### DISCUSSION THREAD

**Discussion by vkomenda** [bionic-gpt/bionic-gpt#433] [Q&A]
I can read the database url with
```
$ kubectl get secret -n bionic-gpt database-urls -o jsonpath='{.data.readonly-url}' | base64 --decode
postgres://bionic_readonly:1353726180174@bionic-db-cluster-rw:5432/bionic-gpt?sslmode=require
```
However just giving it to `psql` doesn't work:
```
$ psql postgres://bionic_readonly:1353726180174@bionic-db-cluster-rw:5432/bionic-gpt?sslmode=require
psql: error: could not translate host name "bionic-db-cluster-rw" to address: Name or service not known
```
How can I connect?

**Comment by vkomenda** [comment]
I see, it's a DNS issue. I have to use the pod IP from, say `kubectl get pods -n bionic-gpt -o yaml`, but the pod is called something else there, `bionic-db-cluster`, without the `-rw`.

**Reply by ianpurton** [reply]
You can use kubectl portforward to access bionic-db-cluster then connect via localhost
